### ISSUE TITLE
IP Leak issue

### ISSUE THREAD

**Issue by moom0o** [TeamPiped/Piped#738]
I noticed piped has the same problem I had with invidious where anyone can get the server's ip address by watching a video, opening developer tools (F12), going to network, and finding a request from the piped proxy.

**Comment by Latrolage** [comment]
Wouldn't you also be able to see the server's ip address by connecting to the server in the first place?

**Comment by moom0o** [comment]
@Latrolage My server is behind cloudflare, which proxies the server to prevent against ddos attacks. Anyone could just find the ip in the network request and bypass Cloudflare.

**Comment by moom0o** [comment]
I assume it would be simple to have the proxy automatically add `&ip=x.x.x.x` to the request to youtube on the backend, and remove it from the client request.

**Comment by FireMasterK** [comment]
> I assume it would be simple to have the proxy automatically add `&ip=x.x.x.x` to the request to youtube on the backend, and remove it from the client request.

It's not since they're different components / applications. I would assume some kind of encryption is required for it.

**Comment by nocturn9x** [comment]
> @Latrolage My server is behind cloudflare, which proxies the server to prevent against ddos attacks. Anyone could just find the ip in the network request and bypass Cloudflare.

Just so you know, CloudFlare is literally a MITM. They listen to **all** of your traffic and unencrypt it before sending it to your origin server. If you really want an Anti-DDoS solution, try something like DDoS guard which has the option to just forward encrypted (or unencrypted for that matter) traffic as-is. Otherwise you're moving away from YouTube just to let Cloudflare track you instead, which doesn't sound like something someone self-hosting an alternative YouTube frontend would want.

In my experience though, the Anti-DDoS of most providers is fine and you don't need services like CloudFlare unless there's people targeting you specifically for some reason.

TL;DR: Don't bother with DDoS protection because 99% of the time no one cares about your little random server
